specification/communication/data-plane/JobRouter/preview/2021-10-20-preview2/communicationservicejobrouter.json

```diff
@@ -0,0 +1,3635 @@
+          "minLength": 0,
+          "type": "string"
+        },
+        "clientId": {
+          "description": "(Optional) Client id, when AppKey is provided\r\nIn context of Azure function, this is usually the name of the key",
+          "maxLength": 500,
+          "minLength": 0,
+          "type": "string"
+        }
+      }
+    },
+    "BestWorkerMode": {
+      "description": "Jobs are distributed to the worker with the strongest abilities available.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DistributionMode"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "readOnly": true
+        },
+        "scoringRule": {
+          "description": "(Optional)\r\nDefine a scoring rule to use, when calculating a score to determine the best worker.\r\nIf not set, will use a default scoring formula that uses the number of job labels that the worker\r\nlabels match, as well as the number of label selectors the worker labels match and/or exceed\r\nusing a logistic function (https://en.wikipedia.org/wiki/Logistic_function).",
+          "x-nullable": true
+        },
+        "scoringRuleOptions": {
+          "$ref": "#/definitions/ScoringRuleOptions"
+        }
+      },
+      "x-ms-discriminator-value": "best-worker"
+    },
+    "CancelExceptionAction": {
+      "description": "An action that marks a job as cancelled",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ExceptionAction"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "readOnly": true
+        },
+        "note": {
+          "description": "(Optional) A note that will be appended to the jobs' Notes collection with th current timestamp.",
+          "maxLength": 2000,
+          "minLength": 0,
+          "type": "string"
+        },
+        "dispositionCode": {
+          "description": "(Optional) Indicates the outcome of the job, populate this field with your own custom values.",
+          "maxLength": 500,
+          "minLength": 0,
+          "type": "string"
+        }
+      },
+      "x-ms-discriminator-value": "cancel"
+    },
+    "CancelJobRequest": {
+      "description": "Request payload for deleting a job",
+      "type": "object",
+      "properties": {
+        "note": {
+          "description": "(Optional) A note that will be appended to the jobs' Notes collection with th current timestamp.",
+          "maxLength": 2000,
+          "minLength": 0,
+          "type": "string"
+        },
+        "dispositionCode": {
+          "description": "Indicates the outcome of the job, populate this field with your own custom values.\r\nIf not provided, default value of \"Cancelled\" is set.",
+          "maxLength": 500,
+          "minLength": 0,
+          "type": "string"
+        }
+      }
+    },
+    "CancelJobResponse": {
+      "type": "object"
```

Comment:
`Schema should have a description or title.`
These values are included in the generated developer documentation and should be included. This is often the first exposure a developer will have with your API. You can get a preview of your developer docs by looking at the [OpenAPI Workflow Bot](https://github.com/Azure/azure-rest-api-specs/pull/17872#issuecomment-1040918300) and finding the URL to your [generated docs](https://review.docs.microsoft.com/en-us/rest/api/azure-rest-preview/?branch=result_0fc831c9-4782-4c01-b3eb-9fad204eda9a&view=azure-rest-preview).

Reply:
Tracked using [issue](https://skype.visualstudio.com/SPOOL/_workitems/edit/2788061)

Reply:
These are placeholder empty requests and response objects to prevent breaking changes in the future if we want to include objects in the body for things like close job

Reply:
Even though they are placeholders, they still need to be documented. The descriptions and titles show up in the generated developer docs that get published, so they are prominently exposed to the people using your APIs.

Reply:
Adding descriptions to CancelJobResponse schema causes OAV Model Validation to give an INVALID_TYPE error.
`{
  code: 'INVALID_TYPE',
  message: 'Expected type  but found type object',
  schemaUrl: '.\\preview\\2021-10-20-preview2\\communicationservicejobrouter.json',
  exampleUrl: 'preview\\2021-10-20-preview2\\examples\\Jobs_RequestJobCancel.json',
  schemaPosition: { line: 787, column: 22 },
  schemaJsonPath: undefined,
  examplePosition: { line: 11, column: 16 },
  exampleJsonPath: '$response.200/body',
  severity: 0,
  source: 'response',
  operationId: 'JobRouter_CancelJobAction',
  level: '\x1B[31merror\x1B[39m'
}`

This example was valid before but now is not:
`{
  "parameters": {
    "endpoint": "https://contoso.westus.communications.azure.com",
    "api-version": "2021-10-20_preview2",
    "id": "8780b28c-7079-4de1-9143-4d369289e958",
    "reclassifyJobRequest": {}
  },
  "responses": {
    "200": {
      "body": {}
    }
  }
}
`
However I'm not sure how to provide a valid response body for the example as the CancelJobResponse object being returned has no properties to provide examples of. Is there a good way to do this? I've been struggling to find any examples.

Reply:
Got some help this morning and have a fix I'll push in a little bit:
`"CancelJobResponse": {
      "description": "Response payload from cancelling a job",
      "type": "object",
      "properties": {}
    },`